toptal/picasso
# [FX-5263] Migrate Button component to TailwindCSS

[FX-5263]

### Description

Final PR to apply all work made to migrate Button component to TailwindCSS+Mui/base

This PR contains also changes for dependant components. Therefore, so many breaking changes.

### How to test

- check precisely the last commit ([ce4bcc70895bfb468a0008663fafa59ec48344e2](https://github.com/toptal/picasso/pull/4265/commits/ce4bcc70895bfb468a0008663fafa59ec48344e2))  about specifying changesets for packages and fixes

- [Temploy](https://picasso.toptal.net/feature/migrate-buttons)
- Use "Staff Portal" to check compatibility and regression: https://github.com/toptal/staff-portal/pull/12589
- Use "Client Portal" to check compatibility and regression: https://github.com/toptal/client-portal/pull/8951

Client Portal is checked. All issues related to Button are fixed. There are still other broken cypress tests related to other components

### Development checks

- [x] Add changeset to **ALL** affected components according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)

**Breaking change**

- N/A codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal: https://github.com/toptal/staff-portal/pull/12589
- [x] test alpha package of Picasso in ClientPortal: https://github.com/toptal/client-portal/pull/8951 

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-5263]: https://toptal-core.atlassian.net/browse/FX-5263?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
